docs/pages/onBeforeRender.page.server.mdx

```diff
@@ -55,6 +56,32 @@ This is where you usually pass information about the authenticated user
 > ```
 
 
+## TypeScript
+
+```ts
+export { onBeforeRender }
+
+import type { OnBeforeRenderAsync } from 'vike/types'
+
+const onBeforeRender: OnBeforeRenderAsync = async (
+  pageContext
+): ReturnType<OnBeforeRenderAsync> => {
+  const response = await fetch("https://api.imdb.com/api/movies/")
+  const { movies } = await response.json()
+  return {
+    pageContext: {
+      movies
```

Comment:
(FYI the data fetching guide recommends setting `pageContext.pageProps.movies`, but I think it's fine if we leave your example as is. The upcoming `+data.js` will simplify all this; I can't wait for us to implement it once the V1 design is released. FYI the `onBeforeRender()` hook will still be useful for advanced integrations such as GraphQL.)

Reply:
:+1: fixed